tool_sideview/sideview_view.py

```diff
@@ -1621,21 +1618,22 @@ def get_layer(spatialite_path, table_name, geom_column=""):
             # geom = line.get('geom', QgsGeometry.fromPolyline([p1, p2]))
 
             feat.setGeometry(geom)
-
+            channel_id = line.get("channel_id", None),
+            # Casting ids to strings is needed due to issue with casting values in memory layers in QGIS < 3.16.6
```

Comment:
Having a comment like this really helps, thanks

Reply:
One additional question: the issue mentions "broken for `>3.16.6`", but it looks like you've changed something for `<3.16.6`. So is the `<` and `>` in the comment the right way around?

Reply:
Actually it's correct, because it seems that there was bug in QGIS < 3.16.6 (in memory data provider) that was allowing to store integers in fields with String type. So pure fix for QGIS > 3.16.6 was breaking plugin for  QGIS < 3.16.6.